sdk/reservations/Azure.ResourceManager.Reservations/src/Custom/Models/ExchangeResult.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.ComponentModel;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Reservations.Models
+{
+    /// <summary> Exchange operation result. </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public partial class ExchangeResult
+    {
+        /// <summary> It should match what is used to GET the operation result. </summary>
+        public ResourceIdentifier Id { get; }
```

Comment:
I am curious why we move the generated property `Id` here? Is it because its type `ResourceIdentifier` is wrong?
**If so**, please also add an `Obsolete` attribute to this property and describe the reason and some guidance to let people use `OperationResultId` instead.
**If the type is not a mistake**, we change its name because we prefer the new name (it is OK and a good thing that makes the SDK better), while we are hiding this property, we should also backing this property to let it return value in the new one, something like this:
```csharp
public ResourceIdentifier Id => new ResourceIdentifier(OperationResultId);
```

Reply:
well I just checked the recording of the corresponding test case, the value of this property is something like `/providers/Microsoft.Capacity/operationResults/2df8adec-a277-46be-b3a0-0fe3bed773ba`
https://github.com/Azure/azure-sdk-for-net/blob/9ff555e3536c1bb1fd4a555485bce120e70eda75/sdk/reservations/Azure.ResourceManager.Reservations/tests/SessionRecords/ExchangeTests/TestCalculateExchangeAndExchange().json#L2733
This is a valid `ResourceIdentifier`. If there is no issues related with this, I suggest we do not change it here.

Reply:
The reason why i did this was because autorest change the type of `id` from `ResourceIdentifier` to `string`, not sure why that happened but now I have decided to revert that change back and specify the format of `ExchangeOperationResultResponse.id` to be `arm-id` so that nothing will be changed about this property